src/test/pages/TeacherRequest.test.tsx

```diff
@@ -0,0 +1,204 @@
+      produce(user, (u) => {
+        u.teacherPending = true;
+      });
+    setupTest({ role: "STUDENT", modifyUser });
+    const message = await screen.findByTestId("teacher-pending-account");
+    expect(message).toHaveTextContent("You already have a teacher upgrade request pending");
+    const form = screen.queryByRole("form");
+    expect(form).not.toBeInTheDocument();
+  });
+
+  it("if student account and no pending request, displays the form with values matching the account", async () => {
+    setupTest({ role: "STUDENT" });
+    const form = await screen.findByRole("form");
+    expect(form).toBeInTheDocument();
+
+    const inputs = [
+      { name: "First name", value: mockUser.givenName },
+      { name: "Last name", value: mockUser.familyName },
+      { name: "Email address", value: mockUser.email },
+      { name: "School", value: mockUser.schoolOther },
+    ];
+
+    inputs.forEach(({ name, value }) => {
+      const input = screen.getByRole("textbox", { name });
+      expect(input).toBeInTheDocument();
+      checkDisabledAndCorrectValue(input, value!);
+    });
+
+    const userVerificationInput = screen.getByLabelText(/URL of a page on your school website/);
+    const otherInformationInput = screen.getByLabelText(/Any other information/);
+    [userVerificationInput, otherInformationInput].forEach((input) => {
+      expect(input).toBeInTheDocument();
+      expect(input).toBeEnabled();
+    });
+  });
+
+  const warningTestCases = [
+    {
+      name: "displays a warning if the email address is unverified, and Submit is disabled",
+      modifyUser: (user: typeof mockUser) =>
+        produce(user, (u) => {
+          u.emailVerificationStatus = "NOT_VERIFIED";
+        }),
+      expectedWarningTestId: "email-unverified",
+    },
+    {
+      name: "displays a warning if the email address is invalid for a teacher account, and Submit is disabled",
+      modifyUser: (user: typeof mockUser) =>
+        produce(user, (u) => {
+          u.email = "exampleteacher@gmail.com";
+        }),
+      expectedWarningTestId: "invalid-email",
+    },
+    {
+      name: "displays a warning if no school is selected, and Submit is disabled",
+      modifyUser: (user: typeof mockUser) =>
+        produce(user, (u) => {
+          u.schoolOther = undefined;
+        }),
+      expectedWarningTestId: "no-school-selected",
+    },
+  ];
+
+  warningTestCases.forEach(({ name, modifyUser, expectedWarningTestId }) => {
+    it(name, async () => {
+      setupTest({ role: "STUDENT", modifyUser });
+      const warning = await screen.findByTestId(expectedWarningTestId);
+      expect(warning).toBeInTheDocument();
+      const submitButton = screen.getByRole("button", { name: "Submit" });
+      expect(submitButton).toBeDisabled();
+    });
+  });
+
+  it("if no Verification Details are specified, form does not submit", async () => {
+    setupTest({ role: "STUDENT" });
+    await screen.findByRole("form");
```

Comment:
You don't seem to be doing anything with this form. Is this here just to ensure that the page has loaded? If so, isn't there another way of doing this? It looks like all you care about here is to wait until the submit button is available, so maybe waiting for that is the right thing to do.